src/legacy/core_plugins/metrics/public/kbn_vis_types/editor_controller.js

```diff
@@ -38,7 +43,7 @@ function ReactEditorControllerProvider(Private, config) {
             vis={this.vis}
             savedObj={this.savedObj}
             timeRange={params.timeRange}
-            renderComplete={() => {}}
+            renderComplete={this.updateVisFields}
```

Comment:
With this, aren't we double rendering the editor component again? (first inside the vis editor on the componentDidMount (when fetching and saving into the state the fields) and another one when the renderComplete method change the props of the viseditor again with these fields)
 
Can't we move the fields fetching directly inside the Editor Controller before rendering the component?
something like reverting these changes https://github.com/elastic/kibana/pull/24832/files#diff-a6dc64277588db05cd6a44af196f7b57L48 and moving `fetchIndexPatternFields` https://github.com/elastic/kibana/pull/24832/files#diff-6d7ae0e7ad72c0a872ead89435838fa3R72 on `EditorController`

Reply:
> With this, aren't we double rendering the editor component again? (first inside the vis editor on the componentDidMount (when fetching and saving into the state the fields) and another one when the renderComplete method change the props of the viseditor again with these fields)

In such case we will fetch it on every render, which happens much more often. Do we really need to update it every time?

Reply:
> With this, aren't we double rendering the editor component again? (first inside the vis editor on the componentDidMount (when fetching and saving into the state the fields) and another one when the renderComplete method change the props of the viseditor again with these fields)

It is the same thing as mutating `props` in React component itself (that how it works before) : `this.props.vis.fields = { ... }`

Reply:
As for me, I'm frustrated that a part of logic doing it by setting new values to `props` object directly, the other part handle it in state (which is different for panel config and visualization). Maybe we need to work on architecture design here?

Reply:
@markov00  What do you think about pass a fetch func to component props ? :
![image](https://user-images.githubusercontent.com/31325372/53243412-d069fc80-36b8-11e9-935e-cf3f84636089.png)